### DISCUSSION TITLE
How could I use Histogram, is there has example to use it?

### DISCUSSION THREAD

**Discussion by helloworlde** [open-telemetry/opentelemetry-java#3112] [Q&A]
Is there has example of how to use every function? It's hardly to find how to use. Thanks very much.

**Comment by jkwatson** [comment]
Hi @helloworlde . Just a few things to clarify first, to make sure we're on the same page.

1) In OpenTelemetry, metrics are still experimental and tagged as alpha, so everything I'm about to write could and probably will change in the future.
2) In OpenTelemetry we distinguish between metric Instruments and Aggregations. Instruments are used to make metric recordings, which are then aggregated before they are sent to an exporter.
3) Every Instrument has a default Aggregation. For example, the ValueRecorder defaults to using a Summary aggregation.
4) Aggregations that are associated with Instruments can be customized via what we call a View. This is somewhat akin to an View from OpenCensus, but not exactly the same.
5) So, if you want to use a Histogram aggregation with one of your instruments, you would do it by registering the appropriate View with the SdkMeterProvider that you configure at application startup.

The API for registering a view is highly experimental. In 1.0.1, this was done by calling the `registerView` method on the SdkMeterPRovider (see javadoc here: https://javadoc.io/doc/io.opentelemetry/opentelemetry-sdk-metrics/latest/io/opentelemetry/sdk/metrics/SdkMeterProvider.html)

However, this is moving to the SdkMeterProviderBuilder class in 1.1.0 which should be released later this week.

So, there is no way to directly use a Histogram. You can only register it as the aggregation for one or more instruments by creating a View (called a AggregationFactory in 1.0.1, btw).

Hope that helps!

**Reply by helloworlde** [reply]
Thanks very much. I want to record request cost time,  could you please help provide an example of how to use it?

And I found the `io.opentelemetry.sdk.metrics.aggregator.AggregatorFactory#histogram` in main branch in [here](https://github.com/open-telemetry/opentelemetry-java/blob/e52969d94c790d6e8865c78bd468326dbea257da/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregator/AggregatorFactory.java#L90), so if the next version can provide histogram directly?

**Reply by jkwatson** [reply]
So, to do that, it sounds like you'd want to use a ValueRecorder and configure a histogram aggregation.

There is no histogram directly in the OpenTelemetry metrics API specification at the moment, and we need to stick with what is specified. That being said, the metrics API specification is in the middle of getting an overhaul. If you're interested in influencing the shape of that API, I recommending joining in the discussion and becoming a part of the metrics Special Interest Group (SIG).  Please see this discussion in the specification repository for starters:  https://github.com/open-telemetry/opentelemetry-specification/discussions/1368
